query-engine/request-handlers/src/graphql/body.rs

```diff
@@ -62,11 +67,11 @@ impl GraphQlBody {
                     .into_iter()
                     .map(|body| GraphQLProtocolAdapter::convert_query_to_operation(&body.query, body.operation_name))
                     .collect();
+                let transaction = bodies
+                    .transaction
```

Comment:
This is a nice idea. But I find it tricky to follow and had to really think about what was happening. 
I think I would prefer:
```
 BatchDocumentTransaction::new(bodies.transaction, bodies.isolation_level)
```
That way the logic is self-contained in one place. If the transaction is `false` we ignore isolation level other wise add the `isolation` level.